### PR TITLE
refacto(tests): fix some eslint-plugin-testing-library errors

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1189

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Truth can only be found in one place: the code

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] test-remi2 - refacto(tests): fix some eslint-plugin-testing-library errors - 2024-06-24T15:56:58Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/57057cdd-d2b4-4592-8375-b94fe12282d2/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] test-remi2 - refacto(tests): fix some eslint-plugin-testing-library errors - 2024-06-24T15:56:58Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/57057cdd-d2b4-4592-8375-b94fe12282d2/logs)

File: libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx [hunk 1]
```diff
@@ -71,29 +72,29 @@ describe('PageSettingsAdvancedFeature', () => {
...
   it('should edit ClusterAdvancedSettings if form is submitted', async () => {
-    const { getByLabelText, getByTestId } = render(<PageSettingsAdvancedFeature />)
+    renderWithProviders(<PageSettingsAdvancedFeature />)
 
     await act(() => {
-      fireEvent.input(getByLabelText('loki.log_retention_in_week'), { target: { value: '2' } })
-      fireEvent.input(getByLabelText('aws.vpc.enable_s3_flow_logs'), { target: { value: 'true' } })
-      fireEvent.input(getByLabelText('load_balancer.size'), { target: { value: '/' } })
-      fireEvent.input(getByLabelText('cloud_provider.container_registry.tags'), {
+      fireEvent.input(screen.getByLabelText('loki.log_retention_in_week'), { target: { value: '2' } })
```
Comment (RemiBonnet) [inline libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx:86]: Can you remove `fireEvent` to be totally iso with our new implementation?

File: libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx [hunk 2]
```diff
@@ -71,29 +72,29 @@ describe('PageSettingsAdvancedFeature', () => {
...
+      fireEvent.input(screen.getByLabelText('load_balancer.size'), { target: { value: '/' } })
+      fireEvent.input(screen.getByLabelText('cloud_provider.container_registry.tags'), {
         target: { value: '{"test":"test"}' },
       })
     })
 
-    expect(getByTestId('submit-button')).toBeEnabled()
+    expect(screen.getByTestId('submit-button')).toBeEnabled()
 
     await act(() => {
```
Comment (RemiBonnet) [inline libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx:96]: Can you remove `act` to be totally iso with our new implementation? or use import not from `'__tests__/utils/setup-jest'` if necessary

File: libs/pages/cluster/src/lib/feature/page-settings-network-feature/crud-modal-feature/crud-modal-feature.spec.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
-import { act, fireEvent, render } from '__tests__/utils/setup-jest'
+import { act, fireEvent } from '__tests__/utils/setup-jest'
```
Comment (RemiBonnet) [inline libs/pages/cluster/src/lib/feature/page-settings-network-feature/crud-modal-feature/crud-modal-feature.spec.tsx:1]: same here

File: libs/pages/cluster/src/lib/feature/page-settings-remote-feature/page-settings-remote-feature.spec.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
-import { act, fireEvent, render } from '__tests__/utils/setup-jest'
+import { act, fireEvent } from '__tests__/utils/setup-jest'
```
Comment (RemiBonnet) [inline libs/pages/cluster/src/lib/feature/page-settings-remote-feature/page-settings-remote-feature.spec.tsx:1]: same here

File: libs/pages/layout/src/lib/feature/dark-mode-enabler/dark-mode-enabler.spec.tsx [hunk 5]
```diff
@@ -1,9 +1,10 @@
-import { getByTestId, render, waitFor } from '__tests__/utils/setup-jest'
+import { waitFor } from '__tests__/utils/setup-jest'
```
Comment (RemiBonnet) [inline libs/pages/layout/src/lib/feature/dark-mode-enabler/dark-mode-enabler.spec.tsx:1]: same here

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx]: As discuss, it causes issues with test not working anymore.
This should be done in a dedicated task.

File: libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/pages/cluster/src/lib/feature/page-settings-advanced-feature/page-settings-advanced-feature.spec.tsx]: As discuss, I've moved imports to `@qovery/shared/util-tests` when possible

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**